From ee17c5619d01d8ea6a5788ee1e56ced8d5a3d74f Mon Sep 17 00:00:00 2001
From: zdohnal <zdohnal@redhat.com>
Date: Thu, 27 Nov 2025 13:32:09 +0100
Subject: [PATCH 25/33] image-tiff.c: Unify return values

TIFF functions used different return values if there is error. Since we don't make difference in error types atm, unify the return value to -1.
---
 cupsfilters/image-tiff.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/cupsfilters/image-tiff.c b/cupsfilters/image-tiff.c
index 748e2db6..df8b6e85 100644
--- a/cupsfilters/image-tiff.c
+++ b/cupsfilters/image-tiff.c
@@ -120,7 +120,7 @@ _cfImageReadTIFF(
     fputs("DEBUG: Images with planar color configuration are not supported!\n", stderr);
     TIFFClose(tif);
     fclose(fp);
-    return (1);
+    return (-1);
   }
 
   if (!TIFFGetField(tif, TIFFTAG_COMPRESSION, &compression))
@@ -143,7 +143,7 @@ _cfImageReadTIFF(
                     "Samples per pixel: %d; Bits per sample: %d\n", samples, bits);
     TIFFClose(tif);
     fclose(fp);
-    return (1);
+    return (-1);
   }
 
   //
@@ -243,7 +243,7 @@ _cfImageReadTIFF(
 		  (unsigned)samples));
     TIFFClose(tif);
     fclose(fp);
-    return (1);
+    return (-1);
   }
 
   //
@@ -365,7 +365,7 @@ _cfImageReadTIFF(
       DEBUG_puts("DEBUG: No enough memory.\n");
       TIFFClose(tif);
       fclose(fp);
-      return (1);
+      return (-1);
     }
     if ((out = (cf_ib_t*)calloc(img->xsize * bpp, sizeof(cf_ib_t))) == NULL)
     {
@@ -373,7 +373,7 @@ _cfImageReadTIFF(
       free(in);
       TIFFClose(tif);
       fclose(fp);
-      return (1);
+      return (-1);
     }
   }
   else
@@ -388,7 +388,7 @@ _cfImageReadTIFF(
       DEBUG_puts("DEBUG: No enough memory.\n");
       TIFFClose(tif);
       fclose(fp);
-      return (1);
+      return (-1);
     }
     if ((out = (cf_ib_t*)calloc(img->ysize * bpp, sizeof(cf_ib_t))) == NULL)
     {
@@ -396,7 +396,7 @@ _cfImageReadTIFF(
       free(in);
       TIFFClose(tif);
       fclose(fp);
-      return (1);
+      return (-1);
     }
   }
 
-- 
2.53.0

